packages/background/src/controllers/NetworkController.ts

```diff
@@ -220,6 +252,42 @@ export default class NetworkController extends BaseController<NetworkControllerS
         return network.name.toUpperCase();
     };
 
+    /**
+     * Sorts the networks by order and adjusts the order property
+     */
+    private orderNetworks = (): void => {
+        const currentNetworks = this.networks;
+
+        // Sort the current networks based on their current order property
+        const orderedNetworks = Object.entries(currentNetworks)
+            .sort(
+                ([, networkValue1], [, networkValue2]) =>
+                    networkValue1.order - networkValue2.order
+            )
+            .reduce(
+                (previousNetworksValue, [networkKey, networkValue]) => ({
+                    ...previousNetworksValue,
+                    [networkKey]: networkValue,
+                }),
+                {}
+            ) as Networks;
+
+        // Adjust order property number to remove gaps
+        let mainnetsCount = 0;
+        let testnetsCount = 0;
+        Object.keys(orderedNetworks).forEach((networkKey) => {
+            if (orderedNetworks[networkKey].test) {
+                orderedNetworks[networkKey].order = testnetsCount;
+                testnetsCount++;
+            } else {
+                orderedNetworks[networkKey].order = mainnetsCount;
+                mainnetsCount++;
+            }
+        });
+
+        this.networks = orderedNetworks;
+    };
+
     public async editNetwork(
```

Comment:
This method should also invoke the order normalizer before doing `this.networks. = newNetworks`.

The `addNetwork` and `deleteNetwork` method too.

Reply:
added it to the edit, although we never edit the order in this function but for the future, if we do
for the delete too. However, for the addNetwork, it doesn't really matter as we always add with the latest order property +1

What do you think?

Reply:
I think that invoking `normalizeNetworksOrder` from the networks setter makes unnecessary to invoke it in every method that edit networks `add` `edit` `remove`